addons/base_import/models/base_import.py

```diff
@@ -637,6 +644,28 @@ def _parse_import_data_recursive(self, model, prefix, data, import_fields, optio
                 # We should be able to manage both case
                 index = import_fields.index(name)
                 self._parse_float_from_data(data, index, name, options)
+            elif field['type'] == 'binary' and field.get('attachment') and ('image' in name or 'icon' in name or 'logo' in name) and name in import_fields:
+                if not self.env.user._can_import_remote_urls():
+                    raise AccessError(_("Only administrators can import images via URL"))
+
+                index = import_fields.index(name)
+                for num, line in enumerate(data):
+                    if re.match(url_regex, line[index]):
+                        try:
+                            with requests.Session() as session:
```

Comment:
That's at entirely the wrong place. The point of using a Session is to do multiple requests through the session such that e.g. the HTTP connection can stay open if we download from the same domain multiple times in a row.


This means the session must be created outside the loop, not inside.